### ISSUE TITLE
Ignore parent directory

### ISSUE THREAD

**Issue by jondashkyle** [jondashkyle/hypha#1]
You usually place your content in a sub-directory of your project. For instance:

```
/project
  /assets
  /content
    /about
      - index.txt
    - index.txt
  /src
```

Because of this, you don’t want `/content` to appear in the path for each of your pages. You can currently get around this by passing a `parent` option.

```js
hypha.readSiteSync('./content', {
  parent: '/content'
})
```

This is fine, however I can’t think of a case where you’d want to ignore anything other than the name of the parent directory, so this should likely just be a `bool` instead of a `str`. Alternatively, could accept both values.

**Comment by lachenmayer** [comment]
I just realised that this option has a totally different interpretation than I imagined, after playing around with it for a while.

My intuition is that the "url" field represents something like a "key in the database" -- the database is the filesystem of course, but that shouldn't matter.

Because of this, I thought this option would _add a prefix_ to each path, relative to the content directory (ie. db root `/`).

Instead, this option actually removes a suffix of the path to the content folder.

I thought that even though the default url behavior was to add `/content`, I could set it to `/` like this:

```js
var hypha = require('.')
var page = hypha.readPageSync('example/content', { parent: '/' })
console.log(page)
```

(You can run this in the Node console from the project directory)

I would have expected the "url" field of the index page to be `/`, and the "about" page to have the url `/about`.

Instead I get this:

```js
{ title: 'Example',
  text: 'yeah *whatsup*',
  name: 'content',
  path: '/example/content',
  url: '/examplecontent',
  files: {},
  pages: 
   { about: 
      { name: 'about',
        path: '/example/content/about',
        url: '/example/content/about',
        source: '/example/content/about' } } }
```

The root page's url isn't what I expected at all, and the about page's url seems to not have changed.

It gets weirder when you put an absolute path in there:

```js
var hypha = require('.')
var path = require('path')
var page = hypha.readPageSync(path.resolve('example/content'))
console.log(page)
```

Gives me

```js
{ title: 'Example',
  text: 'yeah *whatsup*',
  name: 'content',
  path: '/Users/harry/Documents/Code/hypha/example/content',
  url: '/Users/harry/Documents/Code/hypha/example/content',
  files: {},
  pages: 
   { about: 
      { name: 'about',
        path: '/Users/harry/Documents/Code/hypha/example/content/about',
        url: '/Users/harry/Documents/Code/hypha/example/content/about',
        source: '/Users/harry/Documents/Code/hypha/example/content/about' } } }
```

...which is really not what I want at all, especially given that the `path` field has the exact same information.

I feel like it would be best to call this option `urlPrefix` or `prefix`, and to make the behaviour as follows:

- the root page maps to `prefix + '/'`
- every page below the root page maps to `prefix + '+' + relative url to content folder` (ie. in the example, `prefix + '/about'`)

The default value of this proposed option could still be `content/`, or it could be `/`. I personally would prefer `/`, but `content/` is a reasonable default too.

Hope that makes sense :)

**Comment by jondashkyle** [comment]
There are a few compounded bugs here, which should be fixed. Not only that, I believe this can massively improved. First let me explain the intended usage for myself (which might be redundant, but also for future reference if any one else runs into this.)

I’m building a site, and I want the source for all of my content to exist in `/content`.

```
/site
  /assets
  /source
  /content
    /about
    /blog
index.js
```

Since each directory is page of my site, I’d like to use the directory names to generate paths for my router. As all of the content exists in `/content`, my about page path reads `/content/about` if relative to the root. Instead, I’d like to simply remove the parent directory from the beginning of the path, so `/content/about` is `/about` for my router.

I can then use this to easily grab content in my build by simply looking at the key in the object.

```
var content = {
  '/': { },
  '/about': { },
  '/about/jk': { }
}
```

I can also easily map over an array of the keys to build my router. Pretty nice!

So say I have an image associated with a page? If I referenced the `url`, like `/about`, obviously my files don’t actually exist there relative to the root directory. This is what `path` is. Path is the location of the media relative the root directory.

Cool, this is great, but let’s jump further and say we are loading 10 sites over Dat. How do I know what the location of the file relative the source archive? This is what `source` is, but `source` and `path` can probably be the same thing. This was sort of a hack and I forget why I did it this way.

So this looks sort of funny when running a local test and everything is relative to the root directory, but as you move things around—both in the file system, and across a distributed network—it begins to make more sense.

That said, there are a few bugs related to `parent` right now related to absolute/relative paths, etc… These should be fixed, but I think coming up with a solid understanding of expected behavior should happen first. This has so far all been glued together as I’ve been developing the ideas.
